src/main/resources/project/server/createOrUpdateXADataSource.pl

```diff
@@ -400,4 +368,14 @@ sub escape_additional_options {
     $additional_options =~ s|"|\"|gs;
 
     return $additional_options;
+}
+
+sub is_reload_or_restart_required {
+    my $jboss_output = shift;
+    croak "required param is not provided (jboss_output)" unless defined $jboss_output;
+    if ($jboss_output =~ m/process-state:\s(reload|restart)-required/gs
```

Comment:
If back references are not in use, regular expression should be (?:reload|restart). Also, why do we need "g" flag here?

Reply:
used ?: and removed g flag. agree that this is overhead in the code which is not needed